std/array/static_array.d

```diff
@@ -0,0 +1,159 @@
+module std.array.static_array;
+
+import std.traits:CommonType, Unqual;
+
+// @nogc: https://issues.dlang.org/show_bug.cgi?id=18439
+nothrow @safe pure:
+
+/++
+Returns a static array constructed from `a`. The type of elements can be
+specified implicitly (`int[2] a = staticArray(1,2);`) or explicitly
+(`float[2] a = staticArray!float(1,2)`).
+D20180214T203015: The result is an rvalue, therefore uses like
+`foo(staticArray(1, 2, 3)` may be inefficient because of the copies.
++/
+pragma(inline, true) U[T.length] staticArray(U = CommonType!T, T...)(T a) @nogc
+{
+    return [a];
+}
+
+// D20180214T185602: Workaround https://issues.dlang.org/show_bug.cgi?id=16779 (make alias to staticArray once fixed)
+pragma(inline, true) U[T.length] staticArrayCast(U, T...)(T a) @nogc
+{
+    enum n = T.length;
+    U[n] ret = void;
```

Comment:
If `U` is a struct you need to check if it has an elaborate destructor and in that case not void-initialize `ret` because the destructor for `ret[i]` will be run before `a[i]` is blitted over it. An alternative could be to use `emplace`.

Reply:
done, used emplaceRef wherever applicable